File: javascript/src/Behavioral/Mediator/DvdLowercaseTitle.js
(function() {
  var __hasProp = {}.hasOwnProperty,
    __extends = function(child, parent) { for (var key in parent) { if (__hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; };

  define(['Behavioral/Mediator/DvdTitle'], function(DvdTitle) {
    var DvdLowercaseTitle;
    return DvdLowercaseTitle = (function(_super) {
      __extends(DvdLowercaseTitle, _super);

      DvdLowercaseTitle.prototype.LowercaseTitle = null;

      DvdLowercaseTitle.prototype.dvdMediator = null;

      function DvdLowercaseTitle(title, dvdMediator) {
        this.setTitle(title instanceof DvdTitle ? title.getTitle() : title);
        this.resetTitle();
        this.dvdMediator = dvdMediator;
        this.dvdMediator.setLowercase(this);
      }

      DvdLowercaseTitle.prototype.resetTitle = function(title) {
        if (title) {
          this.setTitle(title);
        }
        return this.setLowercaseTitle(this.getTitle().toLowerCase());
      };

      DvdLowercaseTitle.prototype.setSuperTitleLowercase = function() {
        this.setTitle(this.getLowercaseTitle());
        return this.dvdMediator.changeTitle(this);
      };

      DvdLowercaseTitle.prototype.getLowercaseTitle = function() {
        return this.LowercaseTitle;
      };

      DvdLowercaseTitle.prototype.setLowercaseTitle = function(aLowercaseTitle) {
        return this.LowercaseTitle = aLowercaseTitle;
      };

      return DvdLowercaseTitle;

    })(DvdTitle);
  });

}).call(this);

File: go/behavioral/mediator/DvdLowercaseTitle.go
package mediator

import (
	"strings"
)

type DvdLowercaseTitle struct {  
   title string
   LowercaseTitle string
   dvdMediator *DvdMediator
}

func NewDvdLowercaseTitle (title string, med *DvdMediator) *DvdLowercaseTitle {
	obj := new(DvdLowercaseTitle)
	obj.title = title
	obj.ResetTitle()
	obj.dvdMediator = med
	med.SetLowercaseTitle(obj)
	return obj
}
    
func NewDvdLowercaseTitleFromObj (dvdTitle *DvdUpcaseTitle, med *DvdMediator) *DvdLowercaseTitle {
    return  NewDvdLowercaseTitle(dvdTitle.GetTitle(), med)
}       
   
func (this *DvdLowercaseTitle) GetTitle() string {
	return this.title
}

func (this *DvdLowercaseTitle) SetTitle(titleIn string) {
	this.title = titleIn
}

func (this *DvdLowercaseTitle) ResetTitle() {
	 this.setLowercaseTitle( strings.ToLower(this.GetTitle()) )
}

func (this *DvdLowercaseTitle) ResetTitleWithString (title string) {
	 this.SetTitle(title)
	 this.ResetTitle()
}    
   
func (this *DvdLowercaseTitle) SetSuperTitleLowercase() {
	 this.SetTitle(this.GetLowercaseTitle())
	 this.dvdMediator.ChangeLowerTitle(this)
}
   
func (this *DvdLowercaseTitle) GetLowercaseTitle() string {
	return this.LowercaseTitle
}

func (this *DvdLowercaseTitle) setLowercaseTitle(lTitle string) {
    this.LowercaseTitle = lTitle
}

